src/binary/merkle_tree.rs

```diff
@@ -30,34 +31,25 @@ impl<StorageError> From<StorageError> for MerkleTreeError<StorageError> {
 }
 
 #[derive(Debug)]
-pub struct MerkleTree<StorageType> {
+pub struct MerkleTree<TableType, StorageType> {
     storage: StorageType,
     head: Option<Box<Subtree<Node>>>,
     leaves_count: u64,
+    phantom_table: PhantomData<TableType>,
 }
 
-/// The table of the Binary Merkle Tree's nodes. [`MerkleTree`] works with it as
-/// a binary array, where the storage key of the node is the `u64` index and
-/// value is the [`Node`](crate::binary::Node).
-#[derive(Debug)]
-pub struct NodesTable;
-
-impl Mappable for NodesTable {
-    type Key = u64;
-    type SetValue = Primitive;
-    type GetValue = Self::SetValue;
-}
-
-impl<StorageType, StorageError> MerkleTree<StorageType>
+impl<TableType, StorageType, StorageError> MerkleTree<TableType, StorageType>
 where
-    StorageType: StorageMutate<NodesTable, Error = StorageError>,
+    TableType: Mappable<Key = u64, SetValue = Primitive, GetValue = Primitive>,
+    StorageType: StorageMutate<TableType, Error = StorageError>,
```

Comment:
Maybe we want to try:
```rust
    TableTypeKey: AsRef<u64>Mappable<Key = u64, SetValue = Primitive, GetValue = Primitive>,
    TableType::GetValue: AsRef<Primitive>,
    TableType::SetValue: AsRef<Primitive>,
    StorageType: StorageMutate<TableType, Error = StorageError>,
```

With `TryFrom<&Primitive>` from https://github.com/FuelLabs/fuel-merkle/pull/138